atd-vzd/triggers/update_noncr3_location.sql

```diff
@@ -0,0 +1,51 @@
+-- Function for insert and update triggers on atd_apd_blueform
+CREATE OR REPLACE FUNCTION update_noncr3_location()
+  RETURNS TRIGGER 
+  LANGUAGE PLPGSQL
+  AS
+$$
+DECLARE 
+    found_location_id varchar;
+
+BEGIN
+    -- Check if crash is on a major road and of concern to TxDOT.
+    -- NEW.position is recalculated in BEFORE trigger to ensure it is up to date
+    -- when this runs AFTER.
+    IF EXISTS (SELECT ncr3m.*
+                FROM non_cr3_mainlanes AS ncr3m WHERE (
+                            (NEW.position && ncr3m.geometry)
+                            AND ST_Contains(
+                                ST_Transform(
+                                        ST_Buffer(
+                                            ST_Transform(ncr3m.geometry, 2277),
+                                            1,
+                                            'endcap=flat join=round'
+                                        ),
+                                        4326
+                                ), /* transform into 2277 to buffer by a foot, not a degree */
+                                NEW.position
+                            )
+                    )) THEN
+        -- If it is, then set the location_id to None
+		UPDATE atd_apd_blueform SET location_id = NULL 
+        WHERE case_id = NEW.case_id;
+    ELSE 
+        -- If it isn't on a major road and is of concern to Vision Zero, try to find a location_id for it.
```

Comment:
Same on this one. I trusted the existing logic that used to be in a separate function, but I totally get what you are saying here. We briefly touched on how mainlane/non-mainlane (or TxDOT concern/VZ concern) are determined and agreed to keep this existing logic but re-address it. It has been really hard to wrangle the legacy terminology and logic to create a path forward! 😵‍💫

Since we decided to focus on improving how to differentiate TxDOT concern from VZ concern for CR3 crashes, this was punted on to keep things status quo and keep the method of calculating of the non-CR3 crash figures as they are now. 

Also worth noting that the improved method of associating CR3 crashes with major roads (or what we were calling level 5s) relies on columns that only exist for CR3 crashes and not for non-CR3 crashes. This is part of the challenge of figuring out how to improve this part of the code. Maybe we should figure that out now while we are making major changes, though? I need to think about that. I really like your breakdown! We need fresh takes because it has been tough.